pkg/master/master_test.go

```diff
@@ -269,7 +269,7 @@ func TestGetNodeAddresses(t *testing.T) {
 
 	// Fail case (no addresses associated with nodes)
 	nodes, _ := fakeNodeClient.List(metav1.ListOptions{})
-	addrs, err := addressProvider.externalAddresses()
+	addrs, err := addressProvider.getPreferredAddresses([]string{string(api.NodeExternalIP)})
```

Comment:
am still not convinced that we should have all these type cast to drop typing information.

Reply:
The SSH tunnler does specify the interface:
```
type AddressFunc func(preferredAddresses []string) (addresses []string, err error)
```

Imo the ssh tunnler should be be aware of the NodeAddressType.